vermouth/ffinput.py

```diff
@@ -405,16 +404,27 @@ def _base_parser(tokens, context, context_type, section, natoms=None, delete=Fal
         context.interactions[section] = interaction_list
 
 
-def _parse_block_atom(line, context):
-    _, atype, _, resname, name, charge_group, charge = line.split()
+def _parse_block_atom(tokens, context):
+    if tokens[-1].startswith('{'):
+        attributes = _parse_atom_attributes(tokens.pop())
+    else:
+        attributes = {}
+
+    # deque does not support slicing
+    first_six = (tokens.popleft() for i in range(6))
+    _, atype, _, resname, name, charge_group = first_six
     atom = {
         'atomname': name,
         'atype': atype,
         'resname': resname,
-        'charge': float(charge),
         'charge_group': int(charge_group),
     }
-    context.add_atom(atom)
+    # charge and mass are optional, but charge has to be defined for mass to be
```

Comment:
Because charge is column 7 and mass is column 8. Only the 6 first columns are mandatory, but you need a column 7 to have a column 8.

I agree it is weird, but this is something for which I would like to be compatible with the ITP format.

Reply:
Ok, I expected something like this. I agree with your reasoning.